sdv/single_table/base.py

```diff
@@ -822,31 +815,51 @@ def _transform_helper(self, data):
 
         return data
 
-    def preprocess(self, data):
-        """Transform the raw data to numerical space.
+    def _validate_cag(self, data):
+        """Validate the data against the CAG patterns.
 
         Args:
             data (pandas.DataFrame):
-                The raw data to be transformed.
+                The data to validate.
+        """
+        metadata = getattr(self, '_original_metadata', self.metadata)
+        if hasattr(self, '_reject_sampling_patterns'):
+            for pattern in self._reject_sampling_patterns:
+                pattern.validate(data=data, metadata=self._original_metadata)
 
-        Returns:
-            pandas.DataFrame:
-                The preprocessed data.
+        if hasattr(self, '_chained_patterns'):
+            for pattern in self._chained_patterns:
+                pattern.fit(data=data, metadata=metadata)
+                metadata = pattern.get_updated_metadata(metadata)
+                data = pattern.transform(data)
```

Comment:
this code looks very similar to the following. Should we combine them?
https://github.com/sdv-dev/SDV/blob/b6ebea53595679ad6acafbaee37d16f46e335fc1/sdv/single_table/base.py#L813C1-L819C64